src/Microsoft.ML.Core/Utilities/Contracts.cs

```diff
@@ -448,274 +448,293 @@ public static Exception ExceptNotSupp(string msg, params object[] args)
         public static Exception ExceptNotSupp(this IExceptionContext ctx, string msg, params object[] args)
             => Process(new NotSupportedException(GetMsg(msg, args)), ctx);
 
-        // Check - these check a condition and if it fails, throw the corresponding exception.
-        // NOTE: The ordering of arguments to these is standardized to be:
-        // * boolean condition
-        // * parameter name
-        // * parameter value
-        // * message string
-        //
-        // Note that these do NOT support a params array of arguments since that would
-        // involve memory allocation whenever the condition is checked. When message string
-        // args are need, the condition test should be inlined, eg:
-        //   if (!condition)
-        //       throw Contracts.ExceptXxx(fmt, arg1, arg2);
-
-        public static void Check(bool f)
-        {
-            if (!f)
-                throw Except();
-        }
-        public static void Check(this IExceptionContext ctx, bool f)
-        {
-            if (!f)
-                throw Except(ctx);
-        }
-        public static void Check(bool f, string msg)
-        {
-            if (!f)
-                throw Except(msg);
-        }
-        public static void Check(this IExceptionContext ctx, bool f, string msg)
-        {
-            if (!f)
-                throw Except(ctx, msg);
-        }
-
         /// <summary>
-        /// CheckUserArg / ExceptUserArg should be used when the validation of user-provided arguments failed.
-        /// Typically, this is shortly after the arguments are parsed using CmdParser.
+        /// For signalling schema validation issues.
         /// </summary>
-        public static void CheckUserArg(bool f, string name)
-        {
-            if (!f)
-                throw ExceptUserArg(name);
-        }
-        public static void CheckUserArg(this IExceptionContext ctx, bool f, string name)
-        {
-            if (!f)
-                throw ExceptUserArg(ctx, name);
-        }
-        public static void CheckUserArg(bool f, string name, string msg)
-        {
-            if (!f)
-                throw ExceptUserArg(name, msg);
-        }
-        public static void CheckUserArg(this IExceptionContext ctx, bool f, string name, string msg)
-        {
-            if (!f)
-                throw ExceptUserArg(ctx, name, msg);
-        }
+        public static Exception ExceptSchemaMismatch(string paramName, string columnRole, string columnName)
```

Comment:
>ExceptSchemaMismatch [](start = 32, length = 20)

Hi @Zruty0 if we are doing this we ought to change our code analyzer to handle it appropriately. This will involve at least adding the new method name to the target set for the analyzer, as here:

[https://github.com/dotnet/machinelearning/blob/105975b12cfb4f9e5cb6e8209fa7a6a5c31c3eee/tools-local/Microsoft.ML.CodeAnalyzer/ContractsCheckAnalyzer.cs#L80-L86](https://github.com/dotnet/machinelearning/blob/105975b12cfb4f9e5cb6e8209fa7a6a5c31c3eee/tools-local/Microsoft.ML.CodeAnalyzer/ContractsCheckAnalyzer.cs#L80-L86)